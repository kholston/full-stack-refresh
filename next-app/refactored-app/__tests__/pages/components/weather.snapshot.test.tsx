/**
 * @jest-environment jsdom
 */

import {fireEvent, render, screen} from '@testing-library/react'
import PageComponentWeather from '../../../src/pages/components/weather';

describe("PageComponentWeather", () => {
    test('renders correctly', async () => {
        const { container } = render(<PageComponentWeather></PageComponentWeather>)
        expect(container).toMatchSnapshot();
    })

    test('clicks the h1 element and updates the state', async () => {
        const { container } = render(<PageComponentWeather></PageComponentWeather>)

        const header = await screen.findByRole('heading', {level: 1})
        fireEvent.click(header)

        expect(container).toMatchSnapshot();
    })
})
