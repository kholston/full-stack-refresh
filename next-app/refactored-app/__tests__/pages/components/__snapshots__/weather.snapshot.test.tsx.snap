// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageComponentWeather clicks the h1 element and updates the state 1`] = `
<div>
  <h1>
    The weather is 
    sunny
    , and the counter shows 
    2
  </h1>
</div>
`;

exports[`PageComponentWeather renders correctly 1`] = `
<div>
  <h1>
    The weather is 
    sunny
    , and the counter shows 
    1
  </h1>
</div>
`;
